### PR TITLE
[move-binary-format] A few bugfixes

### PR DESCRIPTION
**Pull request by runtian-zhou**
## Motivation

Fixes a few bugs in the move-binary-format resolver.

### Have you read the [Contributing Guidelines on pull requests](https://github.com/diem/diem/blob/main/CONTRIBUTING.md#pull-requests)?

Yes
## Test Plan

Those bugs were exposed in future read-write-set tests. Not sure how we should test them as unit tests for now..

### CODE REVIEW

File: language/move-binary-format/src/layout.rs [hunk 1]
```diff
@@ -132,7 +132,7 @@ impl TypeLayoutBuilder {
             Struct(shi) => MoveTypeLayout::Struct(StructLayoutBuilder::build_from_handle_idx(
                 m,
                 *shi,
-                type_arguments.to_vec(),
+                vec![],
```
Comment (runtian-zhou) [inline language/move-binary-format/src/layout.rs:135]: Here the type arguments readlly should be empty cuz it is a struct instantiation with no type arguements.

File: language/move-binary-format/src/normalized.rs [hunk 2]
```diff
@@ -237,7 +237,7 @@ impl Type {
                 name: name.clone(),
                 type_arguments: type_arguments
                     .iter()
-                    .map(|t| t.subst(type_arguments))
+                    .map(|t| t.subst(type_args))
```
Comment (runtian-zhou) [inline language/move-binary-format/src/normalized.rs:240]: substituted with wrong vector :(

**Review by sblackshear — APPROVED** [review]
Good catches! We should really find a way to test these...

**Pull request merged**